client_cpp/src/posts/body.cc

```diff
@@ -201,20 +195,29 @@ void Post::parse_italics(string_view frag, Post::OnFrag fn)
 
 void Post::parse_reds(string_view frag, Post::OnFrag fn)
 {
-    parse_string(frag, "^r",
+    if (board_config.rb_text) {
+        parse_string(frag, "^r",
         [this, fn](string_view frag) { parse_blues(frag, fn); },
         [this]() {
             wrap_tags(3);
             state.red = !state.red;
         });
+    } else {
+        parse_string(frag, "^r",
```

Comment:
Why even call `parse_string()`, if disabled? Just strip `^r ` and `^b` with `std::string::replace()` and pass to `fn()`. 
An efficient algorithm for this would be: 
1) Check for `^r` and `^b` in `frag`.
  a) If no match, `fn(frag)` and done
2) If matched, copy frag to `std::string`. 
3) Run `std::string::replace()` for both tags.
4) Pass view to modified `std::string` to `fn()`